ciao-controller/instance.go

```diff
@@ -63,17 +64,23 @@ func isCNCIWorkload(workload *types.Workload) bool {
 }
 
 func newInstance(ctl *controller, tenantID string, workload *types.Workload,
-	volumes []storage.BlockDevice, name string) (*instance, error) {
+	volumes []storage.BlockDevice, name string, subnet string) (*instance, error) {
 	id := uuid.Generate()
 
 	if name != "" {
-		existingID, err := ctl.ds.ResolveInstance(tenantID, name)
+		instance, err := ctl.ds.ResolveInstance(tenantID, name)
 		if err != nil {
 			return nil, errors.Wrap(err, "error trying to resolve name")
 		}
 
-		if existingID != "" {
-			return nil, fmt.Errorf("Instance name already in use: %s", name)
+		if instance != nil {
+			// allow repeat names on stopping instances
+			instance.StateLock.RLock()
+			if instance.State != payloads.Stopping {
```

Comment:
This looks like some tedious special casing. Is there any other way we can do that?

Reply:
isn't it nicer though to have a name with some meaning?

Reply:
could still have that "cnci-<tenant-ID>-<subnet>-<random suffix>" or something

Reply:
hah! after all my worrying about the name making sense, the cnci name isn't user visible anyway....
will just make some random thing.